spec/controllers/home_controller_spec.rb

```diff
@@ -2,27 +2,30 @@
 
 RSpec.describe HomeController, type: :controller do
   describe "GET #index" do
-    let(:valid) { true }
-    let!(:github_session) do
-      double(name: 'login', token: 'token', valid?: valid, froggo_path: 'path')
-    end
+    let(:user) { create(:github_user) }
+    let(:github_session) { double(:github_session) }
```

Comment:
Yo dejaría el `let(:valid) { true }` como estaba antes y pasaría los atributos al double para ser más DRY. Fijate que estás repitiendo esto: `allow(github_session).to receive(:valid?) { true }` en cada línea y no hace falta.
En general está bueno definir todas las variables arriba con `let` y luego ir variándolas en contextos. De esta manera, sólo cambias las variables y no repetís la lógica.

Lo otro que haría en estos tests, es probar algo más que `have_http_status(:success)`. Si te fijas los dos casos devuelven `:success`. Tranquilamente podrían estar devolviendo el mismo resultado. Se que sólo probaban esto antes de que llegaras, pero es buen momento para cambiarle.

Reply:
Tienes razón. Cambié los test y agregué que se testeara que el botón del home redirija correctamente: 
![image](https://user-images.githubusercontent.com/6628178/53434965-7332bb80-39d6-11e9-8c09-4f10637989ea.png)

También ahora `GithubAuthenticable` expone `github_user`.